File: 01-02-2024/Question/Python Solutions/Solution3.py
import math

def fare(s,d):#NI-6,HA-3
    path=[800,600,750,900,1400,1200,1100,1500]
    
    stop=['TH','GA','IC','HA','TE','LU','NI','CA']
    
    res=0
    
    st=stop.index(s)#6
    
    ed=stop.index(d)#3
    
    while st!=ed:#7!=3
        st=(st+1)%8#7,0
        res=res+path[st]#path[7],path[0]
        
    return math.ceil(res/200)

s=input()#NI
d=input()#HA

s=s.upper()
d=d.upper()

if s==d:
    print('INVALID INPUT')
else:
    print("%.1f"%fare(s,d),'INR')

File: 01-02-2024/Question/Cpp Solutions/Solution3.cpp
#include <stdio.h>
#include <string.h>
#include <math.h>

int main()
{
    char s[3], d[3];
    scanf("%s%s", s, d);
    char arr[8][3] = {"TH", "GA", "IC", "HA", "TE", "LU", "NI", "CA"};
    float path[8] = {800, 600, 750, 900, 1400, 1200, 1100, 1500};
    float res = 0;
    int i, st, ed;
    for (i = 0; i < 8; i++)
    {
        if (strcmp(strupr(s), arr[i]) == 0)
        {
            st = i;
        }
        if (strcmp(strupr(d), arr[i]) == 0)
        {
            ed = i;
        }
    }
    if (st == ed)
    {
        printf("INVALID INPUT");
        return 0;
    }
    else
    {
        while (st != ed)
        {
            st = (st + 1) % 8;
            res = res + path[st];
        }
        printf("%.1f INR", ceil(res / 200));
    }
}